arch/arm/core/cortex_m/mpu/nxp_mpu.c

```diff
@@ -411,73 +599,48 @@ int arm_core_mpu_buffer_validate(void *addr, size_t size, int write)
  * This function provides the default configuration mechanism for the Memory
  * Protection Unit (MPU).
  */
-static void _nxp_mpu_config(void)
+static int nxp_mpu_init(struct device *arg)
 {
+	ARG_UNUSED(arg);
+
 	u32_t r_index;
 
-	__ASSERT(mpu_config.num_regions <= _get_num_regions(),
-		 "too many static MPU regions defined");
+	if (mpu_config.num_regions > _get_num_regions()) {
+		/* Attempt to configure more MPU regions than
+		 * what is supported by hardware. As this operation
+		 * may be executed during system (pre-kernel) initialization,
+		 * we want to ensure we can detect an attempt to
+		 * perform invalid configuration.
+		 */
+		__ASSERT(0,
+			"Request to configure: %u regions (supported: %u)\n",
+			mpu_config.num_regions,
+			_get_num_regions()
+		);
+		return -1;
```

Comment:
Alright, here, we have an issue:

the mpu_init function returns -1 on error, **but no-one reads this signal**, so we, traditionally have an ASSERT to catch configuration error at Boot time. So, I think this can stay....

Reply:
(Addressed)